torchaudio/pipelines/_wav2vec2/impl.py

```diff
@@ -11,6 +13,34 @@
 __all__ = []
 
 
+class _Wav2Vec2Model(Module):
+    """Wrapper class for :py:class:`~torchaudio.models.Wav2Vec2Model`.
+
+    This is used for layer normalization at the input
+    """
+
+    def __init__(self, model: Wav2Vec2Model):
+        super().__init__()
+        self.model = model
+
+    def forward(self, waveforms: Tensor, lengths: Optional[Tensor] = None) -> Tuple[Tensor, Optional[Tensor]]:
+        waveforms = F.layer_norm(waveforms, waveforms.shape)
+        return self.model(waveforms, lengths)
```

Comment:
okay, `self.model` here + `__getattr__` method will cause infinite recursion. can we rename the model argument to `wav2vec2model`? @mthrok @andreasfloros

Reply:
hmm, it doesn't matter what name it is, the infinite recursion is always there..

Reply:
`__getattr__` is only called if the attribute is not found (https://python-reference.readthedocs.io/en/latest/docs/dunderattr/getattr.html). Since `self.model` is defined in `__init__` I'm not sure why we have recursion

Reply:
I think @nateanl is talking about `__getattr__` in evaluation script. We can discuss that separately.

https://github.com/pytorch/audio/blob/main/examples/asr/librispeech_ctc_decoder/inference.py#L15

Reply:
actually I'm talking about `__getattr__` in this pr. The integration tests failed also because of it https://github.com/pytorch/audio/actions/runs/3569462051/jobs/6001202652

Reply:
oh I see. Why do we need `getattr`? on the wrapper?

Reply:
It seems to due to some magic behind the `nn.Module`'s `__call__` implementation. We do not need generic `getattr` method on the wrapper, don't we?

Reply:
@mthrok to expose the same set of public methods as you mentioned https://github.com/pytorch/audio/pull/2873#discussion_r1034007855. Should I remove it and keep only `extract_features`?

Reply:
I see. Thanks for the explanation.

There are only two public methods `forward` and `extract_features`, so with these methods implemented manually, we do not need generic `getattr`.

https://pytorch.org/audio/main/generated/torchaudio.models.Wav2Vec2Model.html#torchaudio.models.Wav2Vec2Model